### PR TITLE
"Why secretless?" page is updated for consistency

### LINKED ISSUE #422: Website "Why Secretless?" page is consistent with rest of site
Update the "Why Secretless?" page to sync its message with that of the rest of the website

### PR DESCRIPTION
**Pull request by izgeri**
Resolves #422 

Updates "Why Secretless?" page to be more consistent with other language currently being used on the site. This includes the arguments for using Secretless to improve responsiveness to rotation; there is future work to be done to really nail the rotation story, so it's best to remove this content until that later date.

Also makes a minor change to the "Use Open Source" feature on the main page.

### CODE REVIEW

File: docs/docs/overview/why_secretless.md [hunk 1]
```diff
@@ -6,32 +6,36 @@ description: Secretless Broker Documentation
...
+The Secretless Broker is primarily designed to solve two problems. The first is
+**loss or theft of credentials from applications and services**, which can occur by:
 
 - Accidental credential leakage (e.g. credential checked into source control, etc)
 - An attack on a privileged user (e.g. phishing, developer machine compromises, etc)
 - A vulnerability in an application (e.g. remote code execution, environment variable dump, etc)
 
-The second is **downtime caused when applications or services do not respond to credential rotation** and crash or get locked out of target services as a result.
+The second is the **changes to code that interacting with a secrets vault requires,
+and the lack of flexibility that allows**.
```
Comment (sgnn7) [inline docs/docs/overview/why_secretless.md]: I think this sentence needs just a bit of clarity change - took me a couple of re-reads to understand what it was trying to say.

File: docs/docs/overview/why_secretless.md [hunk 2]
```diff
@@ -6,32 +6,36 @@ description: Secretless Broker Documentation
...
 
-The Secretless Broker is specifically written to be really good at interacting with a variety of vaults and abstracting away the security to a specialized module that can easily be audited and expanded with custom plugins. It does a really good job of securing any secrets that it might obtain from a vault, and it does it the same way no matter the developer, technology, or platform. Additionally, it knows what to do when a secret has been changed so that the client users and code never have to respond to changing secrets or even know that they are changing.
+The Secretless Broker is specifically written to be really good at interacting with a variety of vaults and abstracting away the security to a specialized module that can easily be audited and expanded with custom plugins. It does a really good job of securing any secrets that it might obtain from a vault, and it does it the same way no matter the developer, technology, or platform.
 
 In summary, when the client connects to the target service through the Secretless Broker:
 
+- **The client does not have to know how to fetch credentials**
```
Comment (sgnn7) [inline docs/docs/overview/why_secretless.md]: s/`know how to code its APIs`/`know how to interact with it`

File: docs/docs/overview/why_secretless.md [hunk 3]
```diff
@@ -6,32 +6,36 @@ description: Secretless Broker Documentation
...
 
   The client/app no longer has direct access to the password, and therefore cannot reveal it.
 
 - **The client doesn't have to know how to properly manage secrets**
 
   Handling secrets safely involves some complexities, and when every application needs to know how to handle secrets, accidents happen. The Secretless Broker centralizes the client-side management of secrets into one code base, making it easier for developers to focus on delivering features.
-
-- **The client doesn't have to manage secret rotation**
-
-  The Secretless Broker is responsible for establishing connections to the backend, and can handle secrets rotation in a way that's transparent to the client.
```
Comment (sgnn7) [inline docs/docs/overview/why_secretless.md]: Hmm.. why wouldn't we want to highlight the seamless rotation?

Comment (izgeri) [reply docs/docs/overview/why_secretless.md]: it's not all that seamless yet. secretless doesn't pick up when credentials are changed and there is still the possibility of a race condition where it retrieves the cred and then it's changed as it's opening the connection. there is work to be done yet to make this work nicely, and we will do it, and then we can shout it from the rooftops.

File: docs/docs/overview/why_secretless.md (hunk 2 above)
Comment (izgeri) [reply docs/docs/overview/why_secretless.md]: :/ but just before that, it says "it doesn't have to know how to interact with a secrets vault"

File: docs/docs/overview/why_secretless.md (hunk 1 above)
Comment (izgeri) [reply docs/docs/overview/why_secretless.md]: you're not wrong. any suggestions?

Comment (sgnn7) [reply docs/docs/overview/why_secretless.md]: "The second is the need for a client app to know how to interact with secrets vaults" maybe?

File: docs/docs/overview/why_secretless.md (hunk 2 above)
Comment (sgnn7) [reply docs/docs/overview/why_secretless.md]: Hmm.. you're right... maybe just join the sentences? "The client/app no longer has to directly interact with a secrets vault and no code needs to be changed to fetch credentials before opening a connection."
